src/Migrations/M025ReplaceDeprecatedColorsFromContent.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+
+/**
+ * Replace the deprecated pallete colors from content
+ */
+class M025ReplaceDeprecatedColorsFromContent extends MigrationScript
+{
+    /**
+     * Perform the actual migration.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    protected static function execute(MigrationRecord $record): void
+    {
+        global $wpdb;
+
+        $colors = array(
+            [ 'from' => 'grey-80', 'to' => 'grey-900' ],
+            [ 'from' => 'grey-60', 'to' => 'grey-800' ],
+            [ 'from' => 'grey-40', 'to' => 'grey-600' ],
+            [ 'from' => 'grey-20', 'to' => 'grey-500' ],
+            [ 'from' => 'grey-10', 'to' => 'grey-200' ],
+            [ 'from' => 'grey-05', 'to' => 'grey-100' ],
+            [ 'from' => 'gp-green', 'to' => 'green-500' ],
+            [ 'from' => 'dark-blue', 'to' => 'p4-dark-green-800' ],
+            [ 'from' => 'orange-hover', 'to' => 'green-500' ],
+            [ 'from' => 'grey', 'to' => 'grey-900' ],
+            [ 'from' => 'blue', 'to' => 'gp-green-800' ],
+        );
+
+        foreach ($colors as $color) {
+            $from = $color['from'];
+            $to = $color['to'];
+
+            echo sprintf("Replace all the occurrences of '%s' with '%s'\n", $from, $to);
+
+            $sql = '
+                SELECT
+                    ID,
+                    post_parent,
+                    MAX(ID) as latest_post,
+                    count(ID) as total_revisions
+                FROM wp_posts
+                WHERE (
+                    post_content LIKE "%%has-%1$s-background-color%%"
+                    OR post_content LIKE "%%has-%2$s-color%%"
+                )
+                GROUP BY post_parent
+            ';
+
+            $prepared_sql = $wpdb->prepare($sql, [ $from, $from ]);
+            $results = $wpdb->get_results($prepared_sql, ARRAY_A);
+
+            if (count($results) <= 0) {
+                continue;
+            }
+
+            foreach ($results as $post) {
+                $sql = '
+                    UPDATE wp_posts
+                    SET post_content = REPLACE(post_content, "%1$s", "%2$s")
```

Comment:
In a case of a post with one of the background color to change _and_ the name of the color in the article, the text of the article will also be changed:
example here where I added the bg color to an element and the text `amazing blue planet` was changed to `amazing gp-green-800 planet` :grin: 
![Screenshot from 2023-11-08 13-04-46](https://github.com/greenpeace/planet4-master-theme/assets/617346/f0bbe27f-7982-462b-a253-98dfba3d09ca)

For these generic color names (and in general) it's probably better to search and replace the full string `has-...-color` instead. You could generate those strings in your loop and use them for WHERE and REPLACE statements:
```diff
@@ -38,6 +38,11 @@ class M025ReplaceDeprecatedColorsFromContent extends MigrationScript
             $from = $color['from'];
             $to = $color['to'];
 
+            $from_bg = sprintf("has-%s-background-color", $color['from']);
+            $to_bg = sprintf("has-%s-background-color", $color['to']);
+            $from_color = sprintf("has-%s-color", $color['from']);
+            $to_color = sprintf("has-%s-color", $color['to']);
+
             echo sprintf("Replace all the occurrences of '%s' with '%s'\n", $from, $to);
 
             $sql = '
@@ -64,12 +69,16 @@ class M025ReplaceDeprecatedColorsFromContent extends MigrationScript
             foreach ($results as $post) {
                 $sql = '
                     UPDATE wp_posts
-                    SET post_content = REPLACE(post_content, "%1$s", "%2$s")
-                    WHERE (post_content LIKE "%has-%3$s-background-color%" OR post_content LIKE "%has-%4$s-color%")
-                    AND ID=%5$s
-                ';
+                SET post_content = REPLACE(
+                    REPLACE(post_content, "%1$s", "%2$s"),
+                    "%3$s",
+                    "%4$s"
+                )
+                WHERE (post_content LIKE "%%%1$s%%" OR post_content LIKE "%%%3$s%%")
+                AND ID=%5$s
+            ';
 
-                $prepared_sql = $wpdb->prepare($sql, [ $from, $to, $from, $from, $post["ID"]]);
+            $prepared_sql = $wpdb->prepare($sql, [ $from_bg, $to_bg, $from_color, $to_color, $post["ID"]]);
                 $wpdb->query($prepared_sql);
             }
         }
```